lib/resty/balancer/utils.lua

```diff
@@ -43,9 +26,22 @@ do
             end
             return count
         end
+
+    else
+        jit_nkeys = nkeys
     end
 end
-
 _M.nkeys = nkeys
 
+
+function _M.copy(nodes)
+    local newnodes = new_tab(0, jit_nkeys and jit_nkeys(nodes) or 4)
```

Comment:
Can we recycle the new table? Is it on the hot code path?

Reply:
it should not on the hot code path.